packages/e2e-tests/specs/editor/various/preview.test.js

```diff
@@ -248,6 +248,70 @@ describe( 'Preview', () => {
 
 		await previewPage.close();
 	} );
+
+	it( 'should not revert title during a preview right after switching from published to draft', async () => {
+		const editorPage = page;
+
+		// Type Lorem in the title field.
+		await editorPage.type( '.editor-post-title__input', 'Lorem' );
```

Comment:
It would be great if we could avoid relying on developer APIs to select UI features in our tests. It would be preferable to select using elements accessible via [the Accessibility Tree](https://developer.chrome.com/docs/devtools/accessibility/reference/#pane).

In this example, we're selecting the post editor title field by `class` but we could do this in a more accessible way by selecting based on the `aria-label` attribute which is perceivable for users.

The same principle applies to other selections in this test file.

I like to refer to [this order](https://testing-library.com/docs/queries/about#priority) when trying to decide how to select an element. Note that we do not have this library available to us, but we can still follow the general principles using Pupeteer's APIs.

Reply:
Thanks for sharing these useful resources @getdave. I will look into this and update the tests accordingly.

Reply:
This has been addressed in latest commit.